stacks-signer/src/v0/signer.rs

```diff
@@ -891,16 +893,38 @@ impl Signer {
             warn!("{self}: No such block {block_hash}");
             return;
         };
-
-        // record time at which we reached the threshold
-        block_info.signed_group = Some(get_epoch_time_secs());
-        if let Err(e) = self.signer_db.insert_block(&block_info) {
-            error!(
+        // move block to LOCALLY accepted state.
+        // We only mark this GLOBALLY accepted if we manage to broadcast it...
+        if let Err(e) = block_info.mark_locally_accepted(true) {
+            // Do not abort as we should still try to store the block signature threshold
+            warn!("{self}: Failed to mark block as locally accepted: {e:?}");
+        }
+        let _ = self.signer_db.insert_block(&block_info).map_err(|e| {
+            warn!(
                 "Failed to set group threshold signature timestamp for {}: {:?}",
                 block_hash, &e
             );
             panic!("{self} Failed to write block to signerdb: {e}");
-        };
+        });
+        #[cfg(any(test, feature = "testing"))]
```

Comment:
Nit: can you put the fault-injection logic into its own function?

Reply:
Sure, I've gone ahead and moved it to a separate function to help clean things up.

I didn't actually change this code - because of the commit range you were looking at, it included merge commits 🤷🏼 . If you look at the diff for the whole PR, this isn't included.